ui-manchette-with-spacetimechart/src/stories/useElementInView.ts

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect, type RefObject } from 'react';
+
+const MANCHETTE_ACTIONS_HEIGHT = '-40px';
+
+/**
+ * Check if an element is in the viewport of a container
+ * @param ref the element we want to observe
+ * @param containerRef the container we want to observe the element in (needs to be the first scrollable parent)
+ * @param options the options for the intersection observer
+ * - threshold is used to configure how much of the element needs to be visible before the callback is invoked
+ * - rootMargin is used to configure the margin around the root (container) where the intersection observer will trigger
+ * @returns a boolean indicating if the element is in the viewport
+ */
+const useElementInView = <T extends HTMLElement>(
+  ref?: RefObject<T>,
+  containerRef?: RefObject<HTMLElement>,
+  options: IntersectionObserverInit = {
+    threshold: 1,
+    rootMargin: `0px 0px ${MANCHETTE_ACTIONS_HEIGHT} 0px`,
```

Comment:
This hooks sounds like it's generic, but then we're hardcoding details specific to a particular use-cases in the default props. Should we move these defaults to the caller site instead?